aten/src/ATen/native/ScatterGatherChecks.h

```diff
@@ -8,6 +8,28 @@ namespace at { namespace native {
 
 namespace {
 
+// checks whether index.dtype == int64
+// and self.dtyp == src.dtype if src is a Tensor
+static void scatter_gather_dtype_check(
+  const std::string& method_name,
+  const Tensor& self,
+  const Tensor& index,
+  const c10::optional<Tensor>& src_opt = c10::nullopt
+) {
+  TORCH_CHECK_INDEX(
```

Comment:
You need TORCH_CHECK here, not TORCH_CHECK_INDEX?

Reply:
Will change to TORCH_CHECK. Regarding TORCH_INDEX_CHECK, is it only reserved for out of bounds check?

Reply:
Yes, it was erroneously used before to check datatype of index. It will throw python IndexError, which does not make sense if datatype is wrong.